### ISSUE TITLE
DXC generates spurious warning for unintialized vars

### ISSUE THREAD

**Issue by vcsharma** [microsoft/DirectXShaderCompiler#2494]
DXC generates below uninitialized warning when compiling below shader even when structure is initialized.

```
warning: variable 's' is uninitialized when used here
        Foo(s);
            ^
note: initialize the variable 's' to silence this warning
        ST s;
            ^
             = {}
```

```
%dxc -E main -T vs_6_0 %s
struct ST
{
	int a;
};
 
void Foo( out ST s )
{
	s.a = 0;
}
 
ST main() : OUT
{
	ST s;
	Foo(s);
	return s;
}
```

**Comment by vcsharma** [comment]
The uninitialized warnings were enabled in #2424 by enabling the clang diagnostic for uninitialized vars, but looks like it doesn't take into account input modifiers out/inout when a variable/struct is passed to as an function argument.

**Comment by llvm-beanz** [comment]
This is caused by invalid LValueToRValue casts in HLSL call sites. It will be resolved by #5249.

**Comment by damyanp** [comment]
We're not going to address this in DXC - the scale of the change required to do this properly is not worth it when we have clang coming up that will get this right.
